src/dialects/sqlite/query-interface.js

```diff
@@ -155,18 +155,46 @@ export class SqliteQueryInterface extends QueryInterface {
    * @override
    */
   async getForeignKeyReferencesForTable(tableName, options) {
+    const queryOptions = {
+      ...options,
+      type: QueryTypes.FOREIGNKEYS,
+    };
+
     const database = this.sequelize.config.database;
-    const query = this.queryGenerator.getForeignKeysQuery(tableName, database);
-    const result = await this.sequelize.queryRaw(query, options);
-
-    return result.map(row => ({
-      tableName,
-      columnName: row.from,
-      referencedTableName: row.table,
-      referencedColumnName: row.to,
-      tableCatalog: database,
-      referencedTableCatalog: database,
-    }));
+    const query = this.queryGenerator.getForeignKeysQuery(tableName, database, queryOptions);
+    const foreignKeys = await this.sequelize.queryRaw(query, queryOptions);
+
+    // field name translation
+    const changeKey = new Map();
+    changeKey.set('from', 'tableColumnNames');
+    changeKey.set('table', 'referencedTableName');
+    changeKey.set('to', 'referencedTableColumnNames');
+
+    // convert Column field values to array
+    Array.from([foreignKeys]).flat().forEach(foreignKeyObject => {
```

Comment:
this is a lot more complex than what we previously had. Any reason for using this code over the previous one?